pythonPackages.Nikola: 7.8.15 -> 8.0.1

jluttine opened the pull request:
###### Motivation for this change

Upgrade Nikola to 8.0.0 and also add/update some dependencies.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

jluttine committed:
pythonPackages.brotli: init at 1.0.5

jluttine committed:
pythonPackages.Logbook: 1.0.0 -> 1.4.0

jluttine committed:
pythonPackages.Nikola: 7.8.15 -> 8.0.1

Mic92 commented:
@GrahamcOfBorg eval

Mic92 commented on pkgs/development/python-modules/brotli/default.nix:
```diff
@@ -0,0 +1,26 @@
+  checkInputs = [ pytest ];
+
+  checkPhase = ''
+    pytest python/tests
```
Do you know why py.test has both?

jluttine commented:
Should the names of `Nikola` and `Logbook` be normalized to `nikola` and `logbook` or at least made available also with the normalized names? Or do we only normalize names for new package definitions and keep the existing package names inconsistent?

jluttine commented:
Nikola got updated so I updated this pull request so that Nikola is now updated to 8.0.1 instead of 8.0.0.

I'm just unsure should I normalize the package names of Nikola and Logbook as I've been instructed that the names should be normalized but on the other hand the names have these already.

FRidh commented:
We typically avoid renaming existing attributes, though I personally am not against doing the rename.

jluttine commented:
Ok. And I've also understood that it's not ok to have an alias like `nikola = Nikola`, but is that so even in this case when it would keep the backwards compatibility while still providing the normalized name everyone should be expecting? I'm not comfortable breaking the backwards compatibility myself unless there's some consensus on that.

jluttine commented:
So, in my view, this is ready for merging.

jluttine commented:
@FRidh Just a kind reminder in case this has been forgotten. :slightly_smiling_face: In my view, this is ready for merging, but if there's anything I should do, please let me know.
